src/javascript/app/pages/trade/tick_trade.js

```diff
@@ -376,14 +383,18 @@ const TickDisplay = (() => {
                 return;
             }
         }
+
         if (data.tick) {
             spots2  = Tick.spots();
             epoches = Object.keys(spots2).sort((a, b) => a - b);
         } else if (data.history) {
             epoches = data.history.times;
         }
 
-        const has_finished = applicable_ticks && ticks_needed && applicable_ticks.length >= ticks_needed;
+        let has_finished = applicable_ticks && ticks_needed && applicable_ticks.length >= ticks_needed;
+        if (contract_category.match('run')) {
```

Comment:
didn't you set the category to `runs` on line 368?

Reply:
Yeah. but that variable is not in this scope.